logging/metadata/metadata.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metadata provides access to metadata server adoported for needs
+// of the resource detection algorithm. The package partially duplicates
+// implementation of "cloud.google.com/go/compute/metadata" package.
+//
+// This package is a wrapper around the GCE metadata service,
+// as documented at https://developers.google.com/compute/docs/metadata.
+package metadata // import "cloud.google.com/go/logging/metadata"
+
+import (
```

Comment:
A lot of the logic you added here there is already connivence methods for in the compute/metadata package. Can you just use those instead of duplicating things?

Reply:
It is correct. I put a comment explaining why I duplicated the functionality here. The core problem is that customers complain that logging works really slow when used outside of GCP. It happens because the existing code is using compute/metadata implementation with a long timeout and a retrying mechanism. All this is internal to the package. I did not want introducing this change for a special use case. I remembered that the recommended way in go is to duplicate this. The duplication also removed a need for the compute package which, theoretically, should slightly reduce overall size of the dependent packages.

Reply:
as discussed, i will implement a quick validation for metadata server existence and for attribute retrieval i will use the compute/metadata package